### PR TITLE
[RLMObjectStore] Improve the exception message when attempting to migrat...

### PR DESCRIPTION
**Pull request by segiddins**
...e to a bad version number.

Closes https://github.com/realm/realm-cocoa/issues/1475

\c @alazier @tgoyne

### CODE REVIEW

File: Realm/RLMObjectStore.mm [hunk 1]
```diff
@@ -242,7 +242,9 @@ static bool RLMRealmCreateTables(RLMRealm *realm, RLMSchema *targetSchema, bool
 
         // validate versions
         if (oldVersion > newVersion && oldVersion != RLMNotVersioned) {
-            @throw RLMException(@"Version of Realm file on disk is higher than current schema version", @{@"path" : realm.path});
+            NSString *reason = [NSString stringWithFormat:@"Version (%lu) of Realm file on disk (%@) is higher than current schema version (%lu)",
```
Comment (alazier) [inline Realm/RLMObjectStore.mm]: Maybe rephrase to:
`Realm version number %lu is greater than the current schema version %lu for Realm at path '%@'
Seems weird to have the path between the two numbers in the message.

Comment (segiddins) [inline Realm/RLMObjectStore.mm]: done!

**Comment by tgoyne** [comment]
Maybe "Realm at path '%@' has version number %lu greater than current schema version %lu"? Current wording is slightly confusing due to having the path next to the version that doesn't go with the path.

**Comment by alazier** [comment]
::):

**Comment by segiddins** [comment]
done

**Comment by alazier** [comment]
SEVERE ERROR on jenkins - trying to rerun.

**Comment by alazier** [comment]
Do you think it is also worth mentioning 'You must call setSchemaVersion or setDefaultRealmSchemaVersion before accessing an upgraded Realm' or something similar?

**Comment by yoshyosh** [comment]
I think thats very helpful @alazier

**Comment by segiddins** [comment]
@alazier done

**Comment by alazier** [comment]
:+1:
